x/gov/client/rest/utils.go

```diff
@@ -0,0 +1,87 @@
+package rest
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	"github.com/cosmos/cosmos-sdk/client/utils"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/x/gov"
+	"github.com/cosmos/cosmos-sdk/x/gov/tags"
+)
+
+// queryDepositsByTxQuery will query for deposits via a direct txs tags query. It
+// will fetch and build deposits directly from the returned txs and write the
+// JSON response to the provided ResponseWriter.
+//
+// NOTE: SearchTxs is used to facilitate the txs query which does not currently
+// support configurable pagination.
+func queryDepositsByTxQuery(cdc *codec.Codec, cliCtx context.CLIContext, w http.ResponseWriter, proposalID uint64) {
+	tags := []string{
+		fmt.Sprintf("%s='%s'", tags.Action, tags.ActionProposalDeposit),
+		fmt.Sprintf("%s='%s'", tags.ProposalID, []byte(fmt.Sprintf("%d", proposalID))),
```

Comment:
Do we need to `[]byte()` the proposalID here?

Reply:
Yes.